vcx/libvcx/src/credential.rs

```diff
@@ -530,6 +530,27 @@ pub fn get_credential(handle: u32) -> VcxResult<String> {
     })
 }
 
+pub fn delete_credential(handle: u32) -> VcxResult<u32> {
+    HANDLE_MAP.get_mut(handle, |credential| {
+        match credential {
+            Credentials::Pending(_) => {
+                Err(VcxError::from_msg(VcxErrorKind::InvalidCredentialHandle, "Cannot delete credential for Pending object"))
+            }
+            Credentials::V1(_) => {
+                Err(VcxError::from(VcxErrorKind::NotReady))
+            }
+            Credentials::V3(ref credential) => {
+                credential.delete_credential()?;
+                Ok(error::SUCCESS.code_num)
+            }
+        }
+    })
+        .map(|_| error::SUCCESS.code_num)
+        .or(Err(VcxError::from(VcxErrorKind::InvalidCredentialHandle)))
+        .and(release(handle))
+        .and_then(|_| Ok(error::SUCCESS.code_num))
```

Comment:
Why are these lines necessary? How about just
```
Credentials::V3(ref credential) => {
    credential.delete_credential()?;
    Ok(error::SUCCESS.code_num)
}
```

Reply:
I couldn't release the credential object itself in the HANDLE_MAP.get code block. And I have referenced the code from delete_connection(https://github.com/hyperledger/indy-sdk/blob/6ecf3dded79fde74206ec0939d2dda179784768c/vcx/libvcx/src/connection.rs#L958)